### PR TITLE
[DEV-1964]&[DEV-1937]&[DEV-1982] Kyc steps, name check, gs sortColumn

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [x] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [x] Check for linter errors (in PR)
- [x] Test basic user operations
  - Login/logout
  - Registration
  - Update user data
  - Create/delete payment routes
- [x] **Update environment**

#### Post-Release

- Test basic user operations
- Monitor application insights log
- [ ] **Update database** (wallet keys)
- [ ] **Update environment**

### CODE REVIEW

File: migration/1700155481772-addKycLog.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        await queryRunner.query(`CREATE INDEX "IDX_5bb43da53d5e1fed66784bdc7e" ON "kyc_log" ("type") `);
+        await queryRunner.query(`ALTER TABLE "dbo"."user_data" ADD "lastNameCheckDate" datetime2`);
+        await queryRunner.query(`ALTER TABLE "kyc_log" ADD CONSTRAINT "FK_3d1856c90a8b4b20625ce8719e5" FOREIGN KEY ("userDataId") REFERENCES "user_data"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
+        await queryRunner.query(`ALTER TABLE "kyc_log" ADD CONSTRAINT "FK_c6404de74ee43d27c87050939d1" FOREIGN KEY ("bankDataId") REFERENCES "bank_data"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
+    }
```
Comment (davidleomay) [inline migration/1700155481772-addKycLog.js:12]: `"bankDataId" int NOT NULL` doesn't work with inheritance 🤔  it should be nullable.

File: migration/1699993792847-addKycStep.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const { MigrationInterface, QueryRunner } = require("typeorm");
+
+module.exports = class addKycStep1699993792847 {
+    name = 'addKycStep1699993792847'
+
+    async up(queryRunner) {
+        await queryRunner.query(`CREATE TABLE "kyc_step" ("id" int NOT NULL IDENTITY(1,1), "updated" datetime2 NOT NULL CONSTRAINT "DF_8c9edf8d24eeee815ead6e9de30" DEFAULT getdate(), "created" datetime2 NOT NULL CONSTRAINT "DF_b69ad6cda60bffe5421182267eb" DEFAULT getdate(), "name" nvarchar(255) NOT NULL, "status" nvarchar(255) NOT NULL, "sessionId" nvarchar(255), "userDataId" int NOT NULL, CONSTRAINT "PK_fcd6a4863a74f43a3cad2ce5b92" PRIMARY KEY ("id"))`);
+        await queryRunner.query(`ALTER TABLE "kyc_step" ADD CONSTRAINT "FK_04c352bfacb76b730b9e21c5e8e" FOREIGN KEY ("userDataId") REFERENCES "sqldb-dfx-api-dev".."user_data"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
+    }
```
Comment (davidleomay) [inline migration/1699993792847-addKycStep.js:9]: Remove `"sqldb-dfx-api-dev"..`

Comment (davidleomay) [reply migration/1699993792847-addKycStep.js:9]: Or maybe remove the migration completely? We don't use it and there will be some changes.

**Comment by Yannick1712** [comment]
Add dilisense key to prd env

**Review by davidleomay — APPROVED** [review]
See ToDos in PR description

**Pull request merged**